neuralpp/symbolic/expression.py

```diff
@@ -476,10 +495,29 @@ def dict(self) -> Dict[str, Any]:
         pass
 
 
+class AbelianOperation(Constant, ABC):
+    """
+    An Abelian operation is a commutative, associative binary operation with an identity element.
```

Comment:
I had to search up what Abelian meant haha (o´∀`o) Maybe we can add a link to the wiki page for people who never heard of the term before (https://en.wikipedia.org/wiki/Abelian_group). :)